src/ragger/navigator/navigation_scenario.py

```diff
@@ -0,0 +1,115 @@
+    navigation: NavInsID
+    validation: List[NavInsID]
+    pattern: str
+
+    def __init__(self, device: str, use_case: UseCase, approve: bool):
+        if device.startswith("nano"):
+            self.navigation = NavInsID.RIGHT_CLICK
+            self.validation = [NavInsID.BOTH_CLICK]
+            self.pattern = "^Approve$" if approve else "^Reject$"
+
+        elif device.startswith("stax"):
+            if use_case == UseCase.ADDRESS_CONFIRMATION:
+                self.navigation = NavInsID.USE_CASE_REVIEW_TAP
+                if approve:
+                    self.validation = [NavInsID.USE_CASE_ADDRESS_CONFIRMATION_CONFIRM]
+                else:
+                    self.validation = [NavInsID.USE_CASE_ADDRESS_CONFIRMATION_CANCEL]
+                self.pattern = "^Confirm$"
+
+            elif use_case == UseCase.TX_REVIEW:
+                self.navigation = NavInsID.USE_CASE_REVIEW_TAP
+                if approve:
+                    self.validation = [NavInsID.USE_CASE_REVIEW_CONFIRM]
+                else:
+                    self.validation = [
+                        NavInsID.USE_CASE_REVIEW_REJECT, NavInsID.USE_CASE_CHOICE_CONFIRM
+                    ]
+                self.pattern = "^Hold to sign$"
+
+            else:
+                raise NotImplementedError("Unknown use case")
+
+            # Dismiss the result modal in all cases
+            self.validation += [NavInsID.USE_CASE_STATUS_DISMISS]
+
+        else:
+            raise NotImplementedError("Unknown device")
+
+
+class NavigateWithScenario():
+
+    def __init__(self, navigator, device, test_name, screenshot_path):
+        self.navigator = navigator
+        self.device = device
+        self.test_name = test_name
+        self.screenshot_path = screenshot_path
+
+    def _navigate_with_scenario(self,
+                                scenario: NavigationScenarioData,
+                                path: Optional[Path] = None,
+                                test_name: Optional[str] = None,
+                                custom_screen_text: Optional[str] = None,
+                                do_comparison: bool = True):
+        if custom_screen_text is not None:
+            scenario.pattern = custom_screen_text
+
+        if do_comparison:
+            self.navigator.navigate_until_text_and_compare(
+                navigate_instruction=scenario.navigation,
+                validation_instructions=scenario.validation,
+                text=scenario.pattern,
+                path=path if path else self.screenshot_path,
+                test_case_name=test_name if test_name else self.test_name)
+        else:
+            self.navigator.navigate_until_text(
+                navigate_instruction=scenario.navigation,
+                validation_instructions=scenario.validation,
+                text=scenario.pattern,
+                path=path if path else self.screenshot_path,
+                test_case_name=test_name if test_name else self.test_name)
```

Comment:
```suggestion
        if do_comparison:
            path = None
            test_case_name = None
        else:
            if not path:
                path = self.screenshot_path
            if not test_name:
                test_name = self.test_name
        self.navigator.navigate_until_text_and_compare(
            navigate_instruction=scenario.navigation,
            validation_instructions=scenario.validation,
            text=scenario.pattern,
            path=path,
            test_case_name=test_name)
```

Reply:
Not a fan of the behavior of `navigate_until_text_and_compare` that does actually not compare